### DISCUSSION TITLE
CMI/CMIB Join/Leave messages

### DISCUSSION THREAD

**Discussion by collinP93** [Zrips/CMI#5786] [Enhancement]
Hey there,
First off let me say that CMI is a great plugin with a lot of features making it a one stop shop for most everything.
With that being said though I may have missed this feature if it is already a thing, if you wouldn't mind sending me in the right direction that would be great!

Feature request: Custom Join/Leave messages for both Proxy and server.

Description:
This feature would allow players a more personal/custom feeling instead of the default nondescript message of joining the game/server.
Also it gets annoying when you constantly see player joined the game multiple times when they are switching servers inside a network.

Features:
*Let other players know when you joined the server/network in style with placeholders.
*Disable join/leave messages on a per server basis allowing only getting notified when a player joins the network. (Could be enabled/disabled)
*Let players know what server you or someone else is switching from/to  inside the network. (Could be enabled/disabled)
*Join the server like a ghost by silent joins.
*Special join notices/effects when a player with specific perms join.

Examples:
Player 1: (Out of server/network)
Player 2: (out of server + special perms)

player1 has joined the network!
player1 has appeared in the world.
player1 has appeared in the world from from_bungee_server
player1 gave this world up for to_bungee_server
player1 has left the network.
player1 joined the network for the first time!
[Owner] >> player2 appears in the world with a mighty BOOOM!
[Owner] >> player2 appears in the world inconspicuously. Shhh You never saw this...

Thank you for reading and have a nice day!
Hope to hear back soon.

**Comment by SenseiMunchkin** [comment]
There is a solution for custom join messages per server, I'm not sure there is one for proxy. I will outline the steps required. 

# Step 1
Go to your configuration file `config.yml` and at line `1183`, or in the `Messages:` section, the first sub-section will be Login, followed by a Logout section and finally a First Join Message section. Set the `Use:` option to `true` in all three sections.
![image](https://user-images.githubusercontent.com/25995812/123673126-b949f500-d805-11eb-88ca-851476f0c388.png)
(Location of code might be different)

# Step 2
Go to your lang file in the `/Translations` directory. The option `FirstJoin:` will be the message sent at a player's first join. The following options `LogoutCustom:` and `LoginCustom:` will be sent on player Login and player Logout.
![image](https://user-images.githubusercontent.com/25995812/123673203-ce268880-d805-11eb-829b-f467da9f3a9a.png)
(Location of code might be different)


# Step 3
Enjoy! If there's an error in my guide, please reply and I'll get it sorted out and provide the correct method.
